user-tools/README-CV Object Detection.md

```diff
@@ -0,0 +1,155 @@
+  - `imageQuery` - The image describing the object to search for in the current frame.  The image data must be in one of the following formats...
+    - The base64 encoded string of the JPG image data - This must be the entire JPG image file, not just the visible bytes.
+    - A file:// path to a JPG or PNG image - Be careful when using relative paths as these will be interpreted differently in the Editor vs Runtime builds.
+    - A resource:// path to a READABLE Texture2D in one of your project's Resources folders.
+    - A resource:// path to a .bytes TextAsset in one of your project's Resources folders that is a JPG or PNG saved with a .bytes file extension.  This can be used if you do not want Unity to import your image as a Texture.
+  - `textQuery` - The string describing the object to search for in the current frame.
+  - `threshold` - An optional (can be null/undefined) field to accept a returned match from the object detection model. Returned matches with a confidence score less than this threshold are ignored. Currently, this is only supported for usage with `textQuery`.
+  - `withinRect` - An optional (can be null/undefined) field to limit the search area to a specific pixel region of the current frame.  The SDK will linearly transform the supplied `rect` to fit the current resolution using the `screenSize` as the initial reference resolution.
+    - `screenSize` - The reference resolution in pixels which defines the screen space that `rect` is defined within.
+    - `rect` - The position (x=0, y=0 is bottom left) and size (width, height) of the rectangle that must contain the supplied image data.  The values are defined in pixels.
+  - `actions` - The list of mouse actions to take at the center point of the returned `rect`.
+
+### How to create a CVObjectDetection Image Query using base64 Encoding
+
+1. Find the image you want to use as your query and save it as a JPG file (PNG/BMP/etc are NOT supported at this time).
+   - For this example we saved the file as `tree.jpg`
+   - ![Tree](./sample_images/tree.jpg)
+
+2. Use the `encode_jpg_base64.py` or `encode_jpg_base64.sh` script in this directory to encode the JPG as a base64 string.  The output will be written to STDOUT.
+```shell
+ CURRENT_PATH> python encode_jpg_base64.py sample_images/tree.jpg
+/9j/4AAQ...<rest of the base64 encoded image data>...VLj3ieS/9k=
+```
+or
+```shell
+ CURRENT_PATH> ./encode_jpg_base64.sh sample_images/tree.jpg
+/9j/4AAQ...<rest of the base64 encoded image data>...VLj3ieS/9k=
+```
+
+3. Create a new bot segment json file and copy the base64 encoded output as the `imageQuery` of your CVObjectDetection criteria.
+
+
+## Example Segment JSON
+
+### `endCriteria`
+
+```json
+{
+    "name":"CV Object Detection Criteria: Tree",
+    "endCriteria":[
+        {
+            "type":"CVObjectDetection",
+            "description":"Checks for the presence of a tree.",
+            "transient":true,
+            "data":{
```

Comment:
Don't have a super strong feeling here, but some of the reasons i normally go for separate segment examples are...
1. To be able to cut/paste and have it work right away with less of a chance for user oopses trying to remove the ors
2. to show accurate and concise samples (like not including threshold except in the textQuery example
3. for variation.. to be able to show some with withinRect and some without showing how it is optional